execute/factory.go

```diff
@@ -135,6 +136,15 @@ func (p PluginFactory) NewReportingPlugin(
 		return nil, ocr3types.ReportingPluginInfo{}, fmt.Errorf("failed to create token data observer: %w", err)
 	}
 
+	backgroundTokenDataObserver := tokendata.NewBackgroundObserver(
+		logger.Named(p.lggr, "BackgroundObserver"),
+		baseObserver,
```

Comment:
I'm not sure about the execution flow here, if `baseObserver` is part of `backgroundTokenObserver` it means that we will process every token in the background automatically right? Although it's probably a good pattern, I'm afraid of using integrations with Batch APIs efficiently. (e.g. all attestations for particular token resolved in one request)

Let's imagine the following case. We have two different integrations registered in `baseObserver`
* USDC requiring message-by-message processing
* MyFancyToken provides Batch API which can serve all observations within a single HTTP call

<img width="2340" alt="image" src="https://github.com/user-attachments/assets/9b0c7ac0-dde4-49bb-8c29-7383d557c654">


If backgroundTokenDataObserver wraps `baseObserver`, we enforce message-by-message processing for every underlying tokenDataObserver, right? As long as we have only USDC it's not a problem, but the next integration is going to provide Batch API to reduce that pain and I wonder how this implementation suits it. Is there any easy way to adjust it to properly support batching? I was thinking that `backgroundTokenDataObserver` should be on the particular tokenDataObserver instead of wrapping `baseObserver`, what're your thoughts on that? For instance sth like this

<img width="1274" alt="image" src="https://github.com/user-attachments/assets/c2c67345-3eea-46d8-854c-be484247defe">

Reply:
> I was also thinking about a similar approach but unfortunately it doesn't fit well, since the TokenDataObserver interface, accepts msgs and not individual tokens, so the work cannot be sliced evenly.

That was intentional design. Therefore every tokenDataObserver (every integration) can pick which tokens/messages it should process. Therefore you don't need any "parent" orchestrator to know how to it for them

Reply:
Although it will work well when there is always single tokenDataObserver, I think it will require rework right after adding new integration. I wonder how much effort is required to have it properly designed from the day one? I like the idea that right now everything is separate and doesn't require changes to existing code, but on the other hand, I do wonder how much rework this requires
> I tried to sketch this approach in the code but I ended up touching many components, so I felt that it's in the category of premature optimization.

Reply:
I'm also concerned with embedding the base observer into the background task. Token observations could be completely decoupled from the observation logic, which could create a simpler interface with a single responsibility.